### PR TITLE
Adds preliminary list support

### PR DESCRIPTION
**Pull request by subfuzion**
#10

This PR adds initial list support for fetching values from etcd. The interface has a placeholder for Filter functions, but this isn't implemented yet.
## Verification

```
$ ./swarm start etcd
$ make test-storage
```

### CODE REVIEW

File: data/storage/etcd/store_test.go [hunk 1]
```diff
@@ -163,6 +168,77 @@ func TestUpdate(t *testing.T) {
...
+	ctx := context.Background()
+
+	// store everything under amp/foo/
+	key := "foo"
+
+	// this is a "template" object that provides a concrete type for list to unmarshal into
+	obj := &storage.Project{}
+
+	// unlimited ttl
+	ttl := int64(0)
```
Comment (generalhenry) [inline data/storage/etcd/store_test.go]: why not the more ideomatic:
``` go
var ttl int64
```

**Comment by generalhenry** [comment]
:+1:

File: data/storage/etcd/store_test.go (hunk 1 above)
Comment (subfuzion) [inline data/storage/etcd/store_test.go]: I do get the idiom of letting Go initialize things to the [zero value](https://golang.org/ref/spec#The_zero_value), but for my tests I wanted to be explicit. I had been playing with different TTLs in the tests and decided for now to leave at 0 although I'll replace with a reasonable timeout once I'm ready to close the issue. It also matches the way I'm initializing everything else, as well as shows others how the TTL can be initialized to a different value.

**Comment by chrisccoy** [comment]
:+1:

**Pull request merged**